src/main/kotlin/org/jetbrains/research/testspark/helpers/psiHelpers/JavaPsiHelper.kt

```diff
@@ -173,145 +136,313 @@ class JavaPsiHelper : PsiHelperInterface {
+            fullText += psiClass.text
+
+            return fullText
         }
 
-        // check if a class is noted as abstract in the text
-        return psiClass.text.replace(" ", "")
-            .contains("abstractclass${psiClass.name}", ignoreCase = true)
+    override val superClass: PsiClassWrapper? get() = psiClass.superClass?.let { JavaPsiClassWrapper(it) }
+
+    override val virtualFile: VirtualFile get() = psiClass.containingFile.virtualFile
+
+    override val containingFile: PsiFile get() = psiClass.containingFile
+
+    override fun searchSubclasses(project: Project): Collection<PsiClassWrapper> {
+        val scope = GlobalSearchScope.projectScope(project)
+        val query = ClassInheritorsSearch.search(psiClass, scope, false)
+        return query.findAll().map { JavaPsiClassWrapper(it) }
     }
 
+    override fun getInterestingPsiClassesWithQualifiedNames(
+        psiMethod: PsiMethodWrapper,
+    ): MutableSet<PsiClassWrapper> {
+        val interestingMethods = mutableSetOf(psiMethod as JavaPsiMethodWrapper)
+        for (currentPsiMethod in allMethods) {
+            if ((currentPsiMethod as JavaPsiMethodWrapper).isConstructor) interestingMethods.add(currentPsiMethod)
+        }
+        val interestingPsiClasses = mutableSetOf(this)
+        interestingMethods.forEach { methodIt ->
+            methodIt.parameterList.parameters.forEach { paramIt ->
+                PsiTypesUtil.getPsiClass(paramIt.type)?.let { typeIt ->
+                    JavaPsiClassWrapper(typeIt).let {
+                        if (!it.qualifiedName.startsWith("java.")) {
+                            interestingPsiClasses.add(it)
+                        }
+                    }
+                }
+            }
+        }
+
+        return interestingPsiClasses.toMutableSet()
+    }
+
+    val isInterface: Boolean get() = psiClass.isInterface
+
+    val isAbstractClass: Boolean
+        get() {
+            psiClass.containingFile.virtualFile
+            if (psiClass.isInterface) return false
+
+            val methods = PsiTreeUtil.findChildrenOfType(psiClass, PsiMethod::class.java)
+            for (psiMethod: PsiMethod in methods) {
+                if (psiMethod.body == null) {
+                    return true
+                }
+            }
+
+            // check if a class is noted as abstract in the text
+            return psiClass.text.replace(" ", "")
+                .contains("abstractclass${psiClass.name}", ignoreCase = true)
+        }
+
     /**
      * Checks if the constraints on the selected class are satisfied, so that EvoSuite can generate tests for it.
      * Namely, it is not an enum and not an anonymous inner class.
      *
-     * @param psiClass the PSI class of interest
      * @return true if the constraints are satisfied, false otherwise
      */
-    private fun validateClass(psiClass: PsiClass): Boolean {
+    fun isValid(): Boolean {
         return !psiClass.isEnum && psiClass !is PsiAnonymousClass
     }
```

Comment:
The name is uninformative and general: I cannot understand what it checks without looking at the implementation or reading the Javadoc. Here are some names suggested by GPT:

canGenerateTestsForClass
isTestableClass
isClassEligibleForTestGeneration
isClassSuitableForTesting
isClassTestable